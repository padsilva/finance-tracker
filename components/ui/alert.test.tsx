import { render } from "@testing-library/react";

import { Alert, AlertTitle, AlertDescription } from "./alert";

describe("Alert Components", () => {
  it("should render alert variants correctly", () => {
    const { container } = render(
      <>
        <Alert>
          <AlertTitle>Default Alert</AlertTitle>
          <AlertDescription>This is a default alert</AlertDescription>
        </Alert>

        <Alert variant="destructive">
          <AlertTitle>Destructive Alert</AlertTitle>
          <AlertDescription>This is a destructive alert</AlertDescription>
        </Alert>
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render alert with custom className", () => {
    const { container } = render(
      <Alert className="custom-class">
        <AlertTitle className="title-class">Custom Alert</AlertTitle>
        <AlertDescription className="description-class">
          This is a custom styled alert
        </AlertDescription>
      </Alert>,
    );

    expect(container).toMatchSnapshot();
  });
});
