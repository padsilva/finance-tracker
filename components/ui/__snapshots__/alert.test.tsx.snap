// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Components should render alert variants correctly 1`] = `
<div>
  <div
    class="relative w-full rounded-lg border px-4 py-3 text-sm [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-3 [&>svg~*]:pl-7 bg-background text-foreground [&>svg]:text-primary"
    role="alert"
  >
    <h5
      class="font-medium leading-none tracking-tight"
    >
      Default Alert
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      This is a default alert
    </div>
  </div>
  <div
    class="relative w-full rounded-lg border px-4 py-3 text-sm [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-3 [&>svg~*]:pl-7 border-destructive/50 text-destructive dark:border-destructive [&>svg]:text-destructive dark:bg-foreground"
    role="alert"
  >
    <h5
      class="font-medium leading-none tracking-tight"
    >
      Destructive Alert
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed"
    >
      This is a destructive alert
    </div>
  </div>
</div>
`;

exports[`Alert Components should render alert with custom className 1`] = `
<div>
  <div
    class="relative w-full rounded-lg border px-4 py-3 text-sm [&>svg+div]:translate-y-[-3px] [&>svg]:absolute [&>svg]:left-4 [&>svg]:top-3 [&>svg~*]:pl-7 bg-background text-foreground [&>svg]:text-primary custom-class"
    role="alert"
  >
    <h5
      class="font-medium leading-none tracking-tight title-class"
    >
      Custom Alert
    </h5>
    <div
      class="text-sm [&_p]:leading-relaxed description-class"
    >
      This is a custom styled alert
    </div>
  </div>
</div>
`;
